docs/src/api.md

```diff
@@ -0,0 +1,170 @@
+| `SGLD`               | [`Turing.Inference.SGLD`](@ref)               | Stochastic gradient Langevin dynamics                               |
+| `SGHMC`              | [`Turing.Inference.SGHMC`](@ref)              | Stochastic gradient Hamiltonian Monte Carlo                         |
+| `PolynomialStepsize` | [`Turing.Inference.PolynomialStepsize`](@ref) | Returns a function which generates polynomially decaying step sizes |
+| `HMCDA`              | [`Turing.Inference.HMCDA`](@ref)              | Hamiltonian Monte Carlo with dual averaging                         |
+| `NUTS`               | [`Turing.Inference.NUTS`](@ref)               | No-U-Turn Sampler                                                   |
+| `IS`                 | [`Turing.Inference.IS`](@ref)                 | Importance sampling                                                 |
+| `SMC`                | [`Turing.Inference.SMC`](@ref)                | Sequential Monte Carlo                                              |
+| `PG`                 | [`Turing.Inference.PG`](@ref)                 | Particle Gibbs                                                      |
+| `CSMC`               | [`Turing.Inference.CSMC`](@ref)               | The same as PG                                                      |
+| `externalsampler`    | [`Turing.Inference.externalsampler`](@ref)    | Wrap an external sampler for use in Turing                          |
+
+### Variational inference
+
+See the [variational inference tutorial](https://turinglang.org/docs/tutorials/09-variational-inference/) for a walkthrough on how to use these.
+
+| Exported symbol | Documentation                | Description                             |
+|:--------------- |:---------------------------- |:--------------------------------------- |
+| `vi`            | [`AdvancedVI.vi`](@extref)   | Perform variational inference           |
+| `ADVI`          | [`AdvancedVI.ADVI`](@extref) | Construct an instance of a VI algorithm |
+
+### Automatic differentiation types
+
+These are used to specify the automatic differentiation backend to use.
+See the [AD guide](https://turinglang.org/docs/tutorials/docs-10-using-turing-autodiff/) for more information.
+
+| Exported symbol   | Documentation                        | Description                               |
+|:----------------- |:------------------------------------ |:----------------------------------------- |
+| `AutoForwardDiff` | [`ADTypes.AutoForwardDiff`](@extref) | ForwardDiff.jl backend                    |
+| `AutoReverseDiff` | [`ADTypes.AutoReverseDiff`](@extref) | ReverseDiff.jl backend                    |
+| `AutoZygote`      | [`ADTypes.AutoZygote`](@extref)      | Zygote.jl backend                         |
+| `AutoTracker`     | [`ADTypes.AutoTracker`](@extref)     | Tracker.jl backend                        |
+| `AutoTapir`       | [`ADTypes.AutoTapir`](@extref)       | Tapir.jl backend, only for ADTypes >= 1.0 |
+
```

Comment:
should we add `mooncake` (and `enzyme`)?

Reply:
Mooncake depends on https://github.com/TuringLang/Turing.jl/pull/2341, so I guess it depends on which we merge first. But I think this one is easier to merge :D